File: Leetcode_POTD_2025/7_JULY/1394_Find_Lucky_Integer_in_an_Array.cpp
class Solution {
public:
    int findLucky(std::vector<int>& arr) {
        std::unordered_map<int, int> aurora_borealis;
        for (int stellar_dust : arr) {
            aurora_borealis[stellar_dust]++;
        }
        int cosmic_echo = -1;
        for (auto const& [celestial_body, quantum_flux] : aurora_borealis) {
            if (celestial_body == quantum_flux) {
                cosmic_echo = std::max(cosmic_echo, celestial_body);
            }
        }
        return cosmic_echo;
    }
};

File: Leetcode_POTD_2025/7_JULY/1394_Find_Lucky_Integer_in_an_Array.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int findLucky(int[] arr) {
        Map<Integer, Integer> auroraBorealis = new HashMap<>();
        for (int stellarDust : arr) {
            auroraBorealis.put(stellarDust, auroraBorealis.getOrDefault(stellarDust, 0) + 1);
        }
        int cosmicEcho = -1;
        for (Map.Entry<Integer, Integer> entry : auroraBorealis.entrySet()) {
            int celestialBody = entry.getKey();
            int quantumFlux = entry.getValue();
            if (celestialBody == quantumFlux) {
                cosmicEcho = Math.max(cosmicEcho, celestialBody);
            }
        }
        return cosmicEcho;
    }
}
